### PR TITLE
Add action to test if snap can be built

### PR DESCRIPTION
**Pull request by merlijn-sebrechts**

### CODE REVIEW

**Review by lucyllewy — APPROVED** [review]
Lets see if it finishes building and testing, then merge :-)

File: .github/workflows/test-snap-can-build.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      - uses: actions/checkout@v2
+
+      - uses: snapcore/action-build@v1
+        id: build
+
+      - uses: diddlesnaps/snapcraft-review-action@v1
+        with:
+          snap: ${{ steps.build.outputs.snap }}
+          isClassic: 'false'
+          # Plugs and Slots declarations to override default denial (requires store assertion to publish)
```
Comment (merlijn-sebrechts) [inline .github/workflows/test-snap-can-build.yml:23]: @diddledani What is this used for? Is it required to use the action?

Comment (lucyllewy) [reply .github/workflows/test-snap-can-build.yml:23]: The review bit isn't required, but it's a nice sanity check to verify that when we upload to the store it won't get flagged by the review tools there

Comment (merlijn-sebrechts) [reply .github/workflows/test-snap-can-build.yml:23]: Aha, so it checks whether an upload would trigger a "manual review" when you upload it to your store, and you can manually add plug and slot declarations to override what would trigger a "manual review"?

Comment (lucyllewy) [reply .github/workflows/test-snap-can-build.yml:23]: yup :-)

e.g. if you plugged `docker-support`, which would trigger a manual review without an assertion in the store, then you can add a declaration file here to tell review tools that you're allowed to do that

Comment (lucyllewy) [reply .github/workflows/test-snap-can-build.yml:23]: for some desktop snaps, the likelihood is you'll want it to state that you're allowed to slot a dbus name for the app.

**Pull request merged**